runtime/src/main/java/org/corfudb/runtime/view/Layout.java

```diff
@@ -397,17 +396,16 @@ public String asJSONString() {
      * exception at run time.</p>
      *
      * @return a clone of this instance.
-     * @throws CloneNotSupportedException if the object's class does not
-     *                                    support the {@code Cloneable} interface. Subclasses
-     *                                    that override the {@code clone} method can also
-     *                                    throw this exception to indicate that an instance cannot
-     *                                    be cloned.
      * @see Cloneable
      */
     @Override
-    public Object clone() throws CloneNotSupportedException {
-        super.clone();
-        return parser.fromJson(asJSONString(), Layout.class);
+    public Object clone() {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this "clone" implementation; use a copy constructor or copy factory instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2975)

Reply:
done.